### PR TITLE
Add max-in-flight endpoint to the Vidarr server

### PR DESCRIPTION
**Pull request by iainrb**
- Successful test on a local Vidarr server:
`ibancarz@ld5312-ibanca:~/git/vidarr$ curl http://localhost:8088/api/max-in-flight && echo ""
{"timestamp":1614879464581,"workflows":{"bcl2barcode":{"currentInFlight":0,"maxInFlight":20}}}
`

- This is parts 1&2 of the ticket, modifications to Shesmu are next.
- Timestamp is not essential, but might be useful if we want to know when the current-in-flight value is from.
- Modification to `.gitignore` is to ignore emacs temporary files.

### CODE REVIEW

File: vidarr-server/src/main/java/ca/on/oicr/gsi/vidarr/server/MaxInFlightByWorkflow.java [hunk 1]
```diff
@@ -75,6 +76,22 @@ public ConsumableResourceResponse request(
     }
   }
 
+  /**
```
Comment (apmasell) [inline vidarr-server/src/main/java/ca/on/oicr/gsi/vidarr/server/MaxInFlightByWorkflow.java:79]: I've been keeping methods in alphabetical order. If you're IntelliJ, _Code_ -> _Rerrange Code_ should do it.

**Review by apmasell — APPROVED** [review]
Looks good. Thank you!

**Review by avarsava — APPROVED** [review]

**Comment by apmasell** [comment]
Thanks!

**Comment by iainrb** [comment]
You're welcome. :-)

**Pull request merged**